src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -251,15 +275,85 @@ public void DirectoryInfo_Create_NotFound()
             FileSystemRights rights,
             AccessControlType controlType)
         {
-
             DirectorySecurity security = GetDirectorySecurity(sid, rights, controlType);
             VerifyDirectorySecurity(security);
         }
 
         #endregion
 
+        #region FileInfo Create
+
+        [Fact]
+        public void FileInfo_Create_NullFileInfo()
+        {
+            FileInfo info = null;
+            FileSecurity security = new FileSecurity();
+
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Throws<ArgumentNullException>(() => FileSystemAclExtensions.Create(info, FileMode.Create, FileSystemRights.ReadData, FileShare.Read, DefaultBufferSize, FileOptions.None, security));
```

Comment:
We should also validate the parameter name in the exception whenever possible, using AssertExtensions.Throws.

Reply:
Thanks for the suggestion. I'll use that version instead.